flycheck.el

```diff
@@ -7744,7 +7744,12 @@ Otherwise return the previously used cache directory."
 Look up the directory hierarchy from FILE for a directory
 containing a file that matches REGEXP."
   (locate-dominating-file
-   file
+   ;; We must not pass regular file here because
+   ;; `locate-dominating-file' will just pass it into its second
+   ;; lambda argument . But our lambda expects only directories.
+   (if (file-directory-p file)
+       file
+     (file-name-directory file))
```

Comment:
I'm on version `25.2`.

I still don't understand all the reasons why shouldn't `flycheck--locate-dominating-file-matching` be restricted to requiring directories? This way callers can decide whether they have a regular file path at hand or a directory path and call `file-name-directory` appropriately instead of trying to make `flycheck--locate-dominating-file-matching` as generic as possible.